test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/NetCorePackageReferenceProjectTests.cs

```diff
@@ -447,6 +456,1909 @@ public async Task GetInstalledVersion_WithAssetsFile_ChangingPackageSpec_Returns
         }
 
+        [Fact]
+        public async Task TestPacMan_PreviewBuildIntegratedProjectsActionsAsync_InstallPackageForAllProjects_Success()
+        {
+            var projectDirectories = new List<TestDirectory>();
+
+            try
+            {
+                using (var testDirectory = TestDirectory.Create())
+                using (var testSolutionManager = new TestSolutionManager())
+                {
+                    // Set up Package Source
+                    var sources = new List<PackageSource>();
+                        var packageA_Version100 = new SimpleTestPackageContext("packageA", "1.0.0");
+                        var packageB_Version100 = new SimpleTestPackageContext("packageB", "1.0.0");
+                        var packageA = packageA_Version100.Identity;
+                        var packageB = packageB_Version100.Identity;
+                    var packageSource = Path.Combine(testDirectory, "packageSource");
+                    await SimpleTestPackageUtility.CreateFolderFeedV3Async(
+                        packageSource,
+                        PackageSaveMode.Defaultv3,
+                        packageA_Version100,
+                        packageB_Version100
+                        );
+
+                    sources.Add(new PackageSource(packageSource));
+                    var sourceRepositoryProvider = TestSourceRepositoryUtility.CreateSourceRepositoryProvider(sources);
+
+                    // Project
+                    int numberOfProjects = 4;
+                    var projectCache = new ProjectSystemCache();
+                    IVsProjectAdapter projectAdapter = (new Mock<IVsProjectAdapter>()).Object;
+                    ResolutionContext resolutionContext = (new Mock<ResolutionContext>()).Object;
+                    var packageSpecs = new PackageSpec[numberOfProjects];
+                    var projectFullPaths = new string[numberOfProjects];
+                    var deleteOnRestartManager = new TestDeleteOnRestartManager();
+                    var nuGetPackageManager = new NuGetPackageManager(
+                        sourceRepositoryProvider,
+                        NullSettings.Instance,
+                        testSolutionManager,
+                        deleteOnRestartManager);
+
+                    var testNuGetProjectContext = new TestNuGetProjectContext();
+                    var netCorePackageReferenceProjects = new List<NetCorePackageReferenceProject>();
+                    var prevProj = string.Empty;
+                    PackageSpec prevPackageSpec = null;
+
+                    // Create projects
+                    for (var i = numberOfProjects-1; i >= 0; i--)
+                    {
+                        var projectName = $"project{i}";
+                        var directory = TestDirectory.Create();
+                        projectDirectories.Add(directory);
+                        var projectFullPath = Path.Combine(directory.Path, projectName + ".csproj");
```

Comment:
Refer to common on line 571:

> You shouldn't need this. It defeats the purpose of a disposable if you need to dispose of it in a finally block.

The directory here can be:

>                         var projectFullPath = Path.Combine(testDirectory.Path, projectName, projectName + ".csproj");

Reply:
Addressed this issue.